### PR TITLE
Guarded transactions

### PR DESCRIPTION
**Pull request by AdoAdoAdo**
This PR adds support for guarded transactions: 
* the required field transactions
* methods to allow the guardian to apply his signature over the user transaction

### CODE REVIEW

File: builders/txBuilder.go [hunk 1]
```diff
@@ -40,52 +41,103 @@ func NewTxBuilder(txSigner TxSigner) (*txBuilder, error) {
...
-// ApplySignatureAndGenerateTx will apply the corresponding sender and compute the signature field and
+// ApplyUserSignatureAndGenerateTx will apply the corresponding sender and compute the signature field and
 // generate the transaction instance
-func (builder *txBuilder) ApplySignatureAndGenerateTx(
+func (builder *txBuilder) ApplyUserSignatureAndGenerateTx(
 	skBytes []byte,
 	arg data.ArgCreateTransaction,
 ) (*data.Transaction, error) {
-
+	newArg := arg
```
Comment (dragos-rebegea) [inline builders/txBuilder.go:66]: why don`t we use arg directly and make a copy ?

File: builders/txBuilder.go [hunk 2]
```diff
@@ -131,26 +183,47 @@ func transactionToNodeTransaction(tx *data.Transaction) (*transaction.Transactio
...
+		Signature:         signaturesBytes,
+		Options:           tx.Options,
+		GuardianAddr:      guardianAddrBytes,
+		GuardianSignature: guardianSigBytes,
 	}, nil
 }
 
-func (builder *txBuilder) createUnsignedMessage(arg data.ArgCreateTransaction) ([]byte, error) {
-	arg.Signature = ""
+func TransactionToUnsignedTx(tx *data.Transaction) *data.Transaction {
```
Comment (dragos-rebegea) [inline builders/txBuilder.go:216]: this function can be private if we don`t intend to use it anywhere else
```suggestion
func transactionToUnsignedTx(tx *data.Transaction) *data.Transaction {
```

File: data/transaction.go [hunk 3]
```diff
@@ -23,17 +23,19 @@ type SendTransactionsResponse struct {
...
+	Value             string `json:"value"`
+	RcvAddr           string `json:"receiver"`
+	SndAddr           string `json:"sender"`
+	GasPrice          uint64 `json:"gasPrice,omitempty"`
+	GasLimit          uint64 `json:"gasLimit,omitempty"`
+	Data              []byte `json:"data,omitempty"`
+	Signature         string `json:"signature,omitempty"`
+	ChainID           string `json:"chainID"`
+	Version           uint32 `json:"version"`
+	Options           uint32 `json:"options,omitempty"`
```
Comment (dragos-rebegea) [inline data/transaction.go:36]:
```suggestion
	Options           uint32 `json:"options"`
```

File: builders/txBuilder.go (hunk 1 above)
Comment (AdoAdoAdo) [reply builders/txBuilder.go:66]: removed.

File: builders/txBuilder.go (hunk 2 above)
Comment (AdoAdoAdo) [reply builders/txBuilder.go:216]: done

File: data/transaction.go (hunk 3 above)
Comment (AdoAdoAdo) [reply data/transaction.go:36]: We still need omitempty for backwards compatibility

**Review by raduchis — APPROVED** [review]

File: data/transaction.go (hunk 3 above)
Comment (dragos-rebegea) [reply data/transaction.go:36]: :+1:

**Review by dragos-rebegea — APPROVED** [review]

**Pull request merged**